src/cmd/run.go

```diff
@@ -325,7 +325,7 @@ func runCommandWithFallbacks(container string, command []string, emitEscapeSeque
 			logrus.Debugf("%s", arg)
 		}
 
-		exitCode, err := shell.RunWithExitCode("podman", os.Stdin, os.Stdout, nil, execArgs...)
+		exitCode, err := shell.RunWithExitCode("podman", os.Stdin, os.Stdout, os.Stderr, execArgs...)
```

Comment:
I am unsure about this one.  The standard error stream was being intentionally set to `nil`, but not doing that also helps in some cases.

First, the behaviour shown in the commit message is wrong, but don't be upset.  It took me 5 hours before I could make any sense of what's going on, and I am still unsure.   :)

The behaviour before this patch was:
```
$ output=$(toolbox run /etc)
Error: failed to invoke command /etc in container fedora-toolbox-36
$ echo $output

$
```

ie., `output` was no longer being polluted by the error messages from the inner shell.  This was caused by a combination of the previous [commit](https://github.com/containers/toolbox/commit/a22d7821cb8cea134791dcd4643de21ef565d645) that skips the `--tty` flag when the standard output isn't connected a terminal device, and the [fact](https://github.com/containers/podman/issues/16254) that `--tty` connects the standard error stream of the inner process to the standard output of the outer.

Anyway, the standard error stream was being intentionally set to `nil` to suppress unexpected or redundant error messages from Podman or the inner shells, unless `--verbose` was being used.  eg., in this case, the top two lines are a bit redundant and reveal ugly implementation details:
```
output=$(toolbox run /etc)
/bin/sh: line 1: /etc: Is a directory
/bin/sh: line 1: exec: /etc: cannot execute: Is a directory
Error: failed to invoke command /etc in container fedora-toolbox-36
```

Instead, a better error might be something like:
```
Error: failed to invoke command /etc in container fedora-toolbox-36
Is /etc an executable?  Do you have permission to execute it?
```

... but that doesn't require the standard error stream from Podman or the inner shells.

However, having the standard error stream can be useful in some other cases:
```
$ output=$(./builddir/src/toolbox run grep bar /foo)
grep: /foo: No such file or directory
```

... because the error comes directly from the command that the user actually tried to run.  We can't detect it through `podman exec` exit codes and such, and showing it doesn't expose any ugly implementation details.  In fact, suppressing these errors hurt usability.

Reply:
I will think about this some more and try to see if I can figure out a way to have both.

Reply:
One correction after reading Matt's [comment](https://github.com/containers/podman/issues/16254#issuecomment-1287778526) on https://github.com/containers/podman/issues/16254:

*Remember: `podman exec --tty` connects the standard error stream of the inner process to the standard output of the outer.*

> the standard error stream was being intentionally set to nil
> to suppress unexpected or redundant error messages from
> Podman or the inner shells, unless --verbose was being
> used. eg., in this case, the top two lines are a bit redundant
> and reveal ugly implementation details:

We weren't able to suppress the standard error stream from *the inner shells* by setting it to `nil`, when `--tty` was being used, because the use of `--tty` merges the standard error stream of the inner process to the standard output.  So, the behaviour in these cases didn't change:
```
$ toolbox run /etc
/bin/sh: line 1: /etc: Is a directory
/bin/sh: line 1: exec: /etc: cannot execute: Is a directory
Error: failed to invoke command /etc in container fedora-toolbox-36
$ toolbox run grep bar /foo
grep: /foo: No such file or directory
```

The previous [commit](https://github.com/containers/toolbox/commit/a22d7821cb8cea134791dcd4643de21ef565d645) fixed the behaviour when the standard output of `toolbox` wasn't connected to a terminal, by not using `--tty`.  This prevented the standard error streams from the inner shells from polluting the `output`, because, in the absence of `--tty`, the standard error stream of the inner process stayed separate from the standard output.  Unfortunately, since Toolbx wasn't reading the standard error stream, because it was set to `nil`, the error messages completely disappeared after the previous patch.  So, as seen before, the behaviour with the previous commit, but before this patch was:
```
$ output=$(toolbox run /etc)
Error: failed to invoke command /etc in container fedora-toolbox-36
$ echo $output

$ output=$(toolbox run grep bar /foo)
$ echo $output

$
```

This patch brings them back the errors from the inner shells by reading the standard error stream:
```
$ output=$(toolbox run /etc)
/bin/sh: line 1: /etc: Is a directory
/bin/sh: line 1: exec: /etc: cannot execute: Is a directory
Error: failed to invoke command /etc in container fedora-toolbox-36
$ echo $output

$ output=$(toolbox run grep bar /foo)
grep: /foo: No such file or directory
$ echo $output

$
```

Originally, what we were able to suppress was the standard error stream of Podman, unless `--verbose` was being used:
```
$ cd /usr/lib64/gnome-shell  # location only present on the host, not the container
$ toolbox run true
Error: directory /usr/lib64/gnome-shell not found in container fedora-toolbox-36
Using /home/rishi instead
```

Now with this patch, the errors from Podman show up even without `--verbose`:
```
$ cd /usr/lib64/gnome-shell  # location only present on the host, not the container
$ toolbox run true
Error: crun: chdir `/usr/lib64/gnome-shell`: No such file or directory: OCI runtime attempted to invoke a command that was not found
Error: directory /usr/lib64/gnome-shell not found in container fedora-toolbox-36
Using /home/rishi instead
```

Reply:
Given that `podman exec --tty` [connects](https://github.com/containers/podman/issues/16254) the standard error stream of the inner process to the standard output of the outer, we won't be able to selectively suppress the error stream of the inner shells to hide ugly implementation details and only reveal what's relevant to the user.  At least not without some serious contortions like overriding the behaviour of `capsh`, `sh`, etc..

What we can suppress is Podman's standard error stream, and I think we should continue to do that as before (ie., unless `--verbose` is used).